File: Music.Model/Contribution.cs
namespace Music.Model
{
	public class Contribution
	{
		public int Id { get; set; }

		public int TrackId { get; set; }
		public Track Track { get; set; }

		public int ArtistId { get; set; }
		public Artist Artist { get; set; }

		public int ContributionTypeId { get; set; }
		public ContributionType ContributionType { get; set; }
	}
}

File: Music.Web/ClientApp/src/app/models/contribution.ts
import { Artist } from './artist';
import { BaseModel } from './base-model';
import { ContributionType } from './contribution-type';
import { Track } from './track';

export interface Contribution extends BaseModel {
  trackId: number;
  track: Track;
  artistId: number;
  artist: Artist;
  contributionType: ContributionType;
}
